### ISSUE TITLE
Runtime error in current master

### ISSUE THREAD

**Issue by kcalvinalvin** [mit-dci/utreexo#114]
```
leveldb: closed
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/mit-dci/utreexo/cmd/bridgenode.genDels(0x17c, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x9048b9c500000001, ...)
        /home/calvin/go/src/github.com/mit-dci/utreexo/cmd/bridgenode/genproofs.go:300 +0x42b
github.com/mit-dci/utreexo/cmd/bridgenode.genAddDel(0x17c, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x9048b9c500000001, ...)
        /home/calvin/go/src/github.com/mit-dci/utreexo/cmd/bridgenode/genproofs.go:278 +0x8e
github.com/mit-dci/utreexo/cmd/bridgenode.BuildProofs(0xc00709110b, 0x5f8f4a, 0xd, 0x5fa4e5, 0x12, 0xc00045c000, 0x0, 0x0)
        /home/calvin/go/src/github.com/mit-dci/utreexo/cmd/bridgenode/genproofs.go:111 +0x84a
main.main()
        /home/calvin/go/src/github.com/mit-dci/utreexo/cmd/main.go:69 +0x2be
```

I'm currently getting this index range error when tested the ```cmd``` binary on testnet

**Comment by adiabat** [comment]
Huh, not sure what's going on.  I tried it with testnet and mainnet blk/rev files and I can't get this same bug to happen.

Can you try to find what it's crashing on?  Is this right at the beginning, or a bunch of blocks in?  Also some info about what transaction it's crashing on, maybe it's something weird.  

Since the tests all work but it crashes on actual use, it'd be great if you could write a test that causes the same crash.

My guess it's some off-by-1 thing.  In `GenDels()` `txinblock` is decremented and should be the same as the index of the rev transactions, but might not be in some case.  I could add a check there and return an error if there isn't 1 fewer rev tx than regular tx.  But maybe this happens in rev blocks sometimes and it's not an error, so we have to deal with it some other way.

**Comment by ysangkok** [comment]
My logs for testnet:

```
Building offsetfile... blk00181.dat
got non magic bytes 00000000, finishing
Building offsetfile... blk00182.dat
blk00182.dat doesn't exist; done building
tip height 1720819
Has access to forestdata, resuming
Forest leaves: 432909
432909 iterations to do
Done 100000 iterations
Done 200000 iterations
Done 300000 iterations
Done 400000 iterations
Forest height: 19
Done restoring forest
Building Proofs and ttldb...
On block : 190001
On block : 200001
remap forest height 19 -> 20
leveldb: closed
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/mit-dci/utreexo/cmd/bridgenode.genDels(0x31916, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1717ccf900000002, ...)
	/home/janus/go/src/github.com/mit-dci/utreexo/cmd/bridgenode/genproofs.go:300 +0x42b
github.com/mit-dci/utreexo/cmd/bridgenode.genAddDel(0x31916, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1717ccf900000002, ...)
	/home/janus/go/src/github.com/mit-dci/utreexo/cmd/bridgenode/genproofs.go:278 +0x8e
github.com/mit-dci/utreexo/cmd/bridgenode.BuildProofs(0xc00709110b, 0x5fbdb8, 0xd, 0x5fd341, 0x12, 0xc00045c000, 0x0, 0x0)
	/home/janus/go/src/github.com/mit-dci/utreexo/cmd/bridgenode/genproofs.go:111 +0x84a
main.main()
	/home/janus/go/src/github.com/mit-dci/utreexo/cmd/main.go:69 +0x2be
```

**Comment by adiabat** [comment]
OK there was a dumb omission a few lines above that, fixed with a3a9dd673fedd4bb81cb7ecbcba51dae36f3179f

But that still doesn't quite solve the problem, now it should just return an error instead of crashing, which is an improvement, but the error is still there.  Somehow there's a mismatch of rev transactions and actual transactions in the block.  I haven't gotten this problem when running it, but it could be that we aren't reading the rev___.dat files right, or that there's some edge case where this happens.  Will try to download more and reproduce.
